wcfsetup/install/files/ts/WoltLabSuite/Core/Bbcode/Code.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * Highlights code in the Code bbcode.
+ *
+ * @author	Tim Duesterhus
+ * @copyright	2001-2019 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module	WoltLabSuite/Core/Bbcode/Code
+ */
+
+import * as Language from "../Language";
+import * as Clipboard from "../Clipboard";
+import * as UiNotification from "../Ui/Notification";
+import Prism from "../Prism";
+import * as PrismHelper from "../Prism/Helper";
+import PrismMeta from "../prism-meta";
+
+const CHUNK_SIZE = 50;
+
+async function waitForIdle(): Promise<void> {
+  return new Promise(function (resolve, _reject) {
+    if ((window as any).requestIdleCallback) {
+      (window as any).requestIdleCallback(resolve, { timeout: 5000 });
+    } else {
+      setTimeout(resolve, 0);
+    }
+  });
+}
+
+class Code {
+  private codeContainer: HTMLElementTagNameMap["code"];
```

Comment:
I usually resort to `HTMLElement` when there is no specific interface available _and_ there are no fancy properties that we want to access. I had to look up what this type is about, which kind of defeats the purpose of having obvious typings. Not sure about this one.